pkg/nodeagent/pod_controller.go

```diff
@@ -237,101 +244,152 @@ func buildRmdWorkload(pod *corev1.Pod) ([]*intelv1alpha1.RmdWorkload, error) {
 
-			switch {
-			case strings.HasSuffix(field, policyConst):
-				if data != "" {
-					rmdWorkload.Spec.Policy = data
-				}
-			case strings.HasSuffix(field, cacheMinConst):
-				minCache, err := strconv.Atoi(data)
-				if err != nil {
-					continue
-				}
-				rmdWorkload.Spec.Rdt.Cache.Min = minCache
-			case strings.HasSuffix(field, mbaPercentageConst):
-				mbaPercentage, err := strconv.Atoi(data)
-				if err != nil {
-					continue
-				}
-				rmdWorkload.Spec.Rdt.Mba.Percentage = mbaPercentage
-			case strings.HasSuffix(field, mbaMbpsConst):
-				mbaMbps, err := strconv.Atoi(data)
-				if err != nil {
-					continue
-				}
-				rmdWorkload.Spec.Rdt.Mba.Mbps = mbaMbps
-			case strings.HasSuffix(field, pstateRatioConst):
-				if data != "" {
-					rmdWorkload.Spec.Plugins.Pstate.Ratio = data
-				}
-			case strings.HasSuffix(field, pstateMonitoringConst):
-				if data != "" {
-					rmdWorkload.Spec.Plugins.Pstate.Monitoring = data
-				}
-			}
+		rmdWorkload.Spec.Policy = policy
+		rmdWorkload.Spec.Rdt.Cache.Min = minCache
+		rmdWorkload.Spec.Rdt.Mba.Percentage = mbaPercentage
+		rmdWorkload.Spec.Rdt.Mba.Mbps = mbaMbps
+		rmdWorkload.Spec.Plugins.Pstate.Ratio = pstateRatio
+		rmdWorkload.Spec.Plugins.Pstate.Monitoring = pstateMonitoring
 
-		}
 		rmdWorkloads = append(rmdWorkloads, rmdWorkload)
 	}
 	return rmdWorkloads, nil
 }
 
+func getAnnotationInfo(workloadData map[string]string, container corev1.Container) (policy string, minCache int, mbaPercentage int, mbaMbps int, pstateRatio string, pstateMonitoring string) {
+	var minCacheStr, mbaPercentStr, mbaMbpsStr string
+	for field, data := range workloadData {
+		if !strings.HasPrefix(field, container.Name) {
+			continue
+		}
+		switch {
+		case strings.HasSuffix(field, policyConst):
+			if data != "" {
+				policy = data
+			}
+		case strings.HasSuffix(field, cacheMinConst):
+			if data != "" {
+				minCacheStr = data
+			}
+		case strings.HasSuffix(field, mbaPercentageConst):
+			if data != "" {
+				mbaPercentStr = data
+			}
+		case strings.HasSuffix(field, mbaMbpsConst):
+			if data != "" {
+				mbaMbpsStr = data
+			}
+		case strings.HasSuffix(field, pstateRatioConst):
+			if data != "" {
+				pstateRatio = data
+			}
+		case strings.HasSuffix(field, pstateMonitoringConst):
+			if data != "" {
+				pstateMonitoring = data
+			}
+		}
+
+	}
+	minCache, err := strconv.Atoi(minCacheStr)
+	checkError(err)
+
+	mbaPercentage, err = strconv.Atoi(mbaPercentStr)
+	checkError(err)
+
+	mbaMbps, err = strconv.Atoi(mbaMbpsStr)
+	checkError(err)
+
+	return policy, minCache, mbaPercentage, mbaMbps, pstateRatio, pstateMonitoring
+}
+
+func checkError(err error) (result string) { //might remove this function in future
+	if err != nil { //only used 3 times
+		return err.Error()
+	}
+	return ""
+}
+
+func getContainerInfo(pod *corev1.Pod, container corev1.Container) containerInformation {
```

Comment:
If we are to explore the idea of using this new struct then this function should return the struct and an error. It is often good practice to return an error (and very common in the Operator pattern) in order to allow you to exit early from the calling function/loop without having to examine the accompanying data (in this case the containerInfo struct). 
So this should look like:

````
func getContainerInfo(pod *corev1.Pod, container corev1.Container) (containerInformation, error) {
````